packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -0,0 +1,321 @@
+        finder = FinderInterface(_finder);
+        require(_getCollateralWhitelist().isOnWhitelist(address(_collateral)), "bond token not supported");
+        collateral = IERC20(_collateral);
+        bond = _bond;
+        rules = _rules;
+        require(_getIdentifierWhitelist().isIdentifierSupported(_identifier), "identifier not supported");
+        identifier = _identifier;
+        require(_liveness > 0, "liveness can't be 0");
+        liveness = _liveness;
+        setAvatar(_owner);
+        setTarget(_owner);
+        transferOwnership(_owner);
+        _sync();
+
+        emit OptimisticGovernorDeployed(_owner, avatar, target);
+    }
+
+    function priceProposed(
+        bytes32 _identifier,
+        uint32 _timestamp,
+        bytes memory _ancillaryData,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) external {
+        emit PriceProposed(_identifier, _timestamp, _ancillaryData, _request);
+    }
+
+    function setBond(uint256 _bond) public onlyOwner {
+        // Value of the bond required for proposals, in addition to the final fee.
+        bond = _bond;
+    }
+
+    function setCollateral(IERC20 _collateral) public onlyOwner nonReentrant {
+        // ERC20 token to be used as collateral (must be approved by UMA Store contract).
+        require(_getCollateralWhitelist().isOnWhitelist(address(_collateral)), "bond token not supported");
+        collateral = _collateral;
+    }
+
+    function setRules(string memory _rules) public onlyOwner {
+        // Set reference to the rules for the avatar (e.g. an IPFS hash or URI).
+        rules = _rules;
+    }
+
+    function setLiveness(uint64 _liveness) public onlyOwner {
+        // Set liveness for disputing proposed transactions.
+        require(_liveness > 0, "liveness can't be 0");
+        liveness = _liveness;
+    }
+
+    function setIdentifier(bytes32 _identifier) public onlyOwner {
+        // Set identifier which is used along with the rules to determine if transactions are valid.
+        require(_getIdentifierWhitelist().isIdentifierSupported(_identifier), "identifier not supported");
+        identifier = _identifier;
+    }
+
+    function sync() public nonReentrant {
+        // Sync the store and oracle contract addresses as well as the final fee.
+        _sync();
+    }
+
+    function proposeTransactions(Transaction[] memory _transactions, bytes memory _explanation) public nonReentrant {
+        // note: Optional explanation explains the intent of the transactions to make comprehension easier.
+        uint256 id = prevProposalId + 1;
+        uint256 time = getCurrentTime();
+        address proposer = msg.sender;
+
+        // Create proposal in memory to emit in an event.
+        Proposal memory proposal;
+        proposal.requestTime = time;
+
+        // Construct the ancillary data.
+        bytes memory ancillaryData = _explanation;
```

Comment:
Probably better for a follow-up, but it seems a little odd that the caller gets to set most/all of the info in the ancillary data. I would worry about this potentially being misleading to voters, as ancillary data is often set by the contract/protocol and is assumed to be tamper-proof.

Might be worth a quick chat with other folks on the team.